src/Handlers/CardsPaymentMethodHandler.php

```diff
@@ -203,7 +203,11 @@ public function pay(
             );
         }
 
-        return $this->paymentResponseHandler->handlePaymentResponse($response, $transaction, $salesChannelContext);
+        $orderNumber = $transaction->getOrder()->getOrderNumber();
+
+        $result = $this->paymentResponseHandler->handlePaymentResponse($response, $orderNumber, $salesChannelContext);
+
+        return $this->paymentResponseHandler->handleShopwareAPIs($transaction, $salesChannelContext, $result);
```

Comment:
Just for consistency with the second function
```suggestion
        return $this->paymentResponseHandler->handleShopwareApis($transaction, $salesChannelContext, $result);
```